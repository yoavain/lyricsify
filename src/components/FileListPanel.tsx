import type { FC } from "react";
import React from "react";
import type { Theme } from "@material-ui/core";
import { createStyles, List, makeStyles } from "@material-ui/core";
import type { RowData } from "./FilePanel";
import FilePanel from "./FilePanel";

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            maxWidth: 360,
            backgroundColor: theme.palette.background.paper
        },
        inline: {
            display: "inline"
        },
        large: {
            width: theme.spacing(6),
            height: theme.spacing(6)
        }
    })
);

interface FileListPanelProps {
    rows: RowData[];
    selectedIndex: number;
    onSelectItemClick: (selectedIndex: number) => void;
    onSaveLyrics: (selectedIndex: number) => void;
}

export const FileListPanel: FC<FileListPanelProps> = (props: FileListPanelProps) => {
    const classes = useStyles();
    const { rows, selectedIndex } = props;

    const handleListItemClick = (event: React.MouseEvent<HTMLDivElement, MouseEvent>, index: number) => {
        props.onSelectItemClick(index);
    };

    const handleSaveLyrics = (event: React.MouseEvent<HTMLDivElement, MouseEvent>, index: number) => {
        props.onSaveLyrics(index);
    };

    return (
        <List className={classes.root}>
            {rows.map((row: RowData, index) => {
                return (
                    <FilePanel
                        key={index}
                        row={row}
                        isSelected={index === selectedIndex}
                        isLast={index === rows.length - 1}
                        onClick={(event) => handleListItemClick(event, index)}
                        onSaveLyrics={(event) => handleSaveLyrics(event, index)}
                        classes={classes}
                    />
                );
            })}
        </List>
    );
};
