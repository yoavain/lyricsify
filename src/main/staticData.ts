export interface RowData {
    filename: string;
    path: string;
    tag: string;
    title: string;
    artist: string;
    album: string;
    track: number;
    year: number;
    length: number;
    size: number;
    lastModified: number;
    hasLyrics: boolean;
}


function createData(filename: string, path: string, tag: string, title: string, artist: string, album: string, track: number, year: number, length: number, size: number, lastModified: number, hasLyrics: boolean = false): RowData {
    return { filename, path, tag, title, artist, album, track, year, length, size, lastModified, hasLyrics };
}

export const rows = [
    createData("01 - Happens To The Heart.flac", "D:\\Music\\Leonard Cohen\\2019 - Thanks For The Dance\\", "FLAC", "Happens To The Heart", "Leonard Cohen", "Thanks For The Dance", 1, 2019, 273, 22749256, 1574467200000),
    createData("02 - Moving On.flac", "D:\\Music\\Leonard Cohen\\2019 - Thanks For The Dance\\", "FLAC", "Moving On", "Leonard Cohen", "Thanks For The Dance", 2, 2019, 192, 14556156, 1574467200000),
    createData("03 - The Night of Santiago.flac", "D:\\Music\\Leonard Cohen\\2019 - Thanks For The Dance\\", "FLAC", "The Night of Santiago", "Leonard Cohen", "Thanks For The Dance", 3, 2019, 255, 20998200, 1574467200000),
    createData("04 - Thanks For The Dance.flac", "D:\\Music\\Leonard Cohen\\2019 - Thanks For The Dance\\", "FLAC", "Thanks For The Dance", "Leonard Cohen", "Thanks For The Dance", 4, 2019, 253, 18920215, 1574467200000),
    createData("05 - It's Torn.flac", "D:\\Music\\Leonard Cohen\\2019 - Thanks For The Dance\\", "FLAC", "It's Torn", "Leonard Cohen", "Thanks For The Dance", 5, 2019, 178, 13998876, 1574467200000),
    createData("06 - The Goal.flac", "D:\\Music\\Leonard Cohen\\2019 - Thanks For The Dance\\", "FLAC", "The Goal", "Leonard Cohen", "Thanks For The Dance", 6, 2019, 72, 5576134, 1574467200000),
    createData("07 - Puppets.flac", "D:\\Music\\Leonard Cohen\\2019 - Thanks For The Dance\\", "FLAC", "Puppets", "Leonard Cohen", "Thanks For The Dance", 7, 2019, 160, 12369996, 1574467200000),
    createData("08 - The Hills.flac", "D:\\Music\\Leonard Cohen\\2019 - Thanks For The Dance\\", "FLAC", "The Hills", "Leonard Cohen", "Thanks For The Dance", 8, 2019, 258, 25574353, 1574467200000),
    createData("09 - Listen To The Hummingbird.flac", "D:\\Music\\Leonard Cohen\\2019 - Thanks For The Dance\\", "FLAC", "Listen To The Hummingbird", "Leonard Cohen", "Thanks For The Dance", 9, 2019, 121, 7459732, 1574467200000)
];