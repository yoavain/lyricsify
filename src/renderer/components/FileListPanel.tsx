import * as React from "react";
import makeStyles from "@material-ui/core/styles/makeStyles";
import { createStyles, Theme } from "@material-ui/core";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import Avatar from "@material-ui/core/Avatar";
import ListItemText from "@material-ui/core/ListItemText";
import Typography from "@material-ui/core/Typography";
import Divider from "@material-ui/core/Divider";
import { RowData } from "../../../test/resources/staticData";

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            maxWidth: 360,
            backgroundColor: theme.palette.background.paper
        },
        inline: {
            display: "inline"
        },
        large: {
            width: theme.spacing(6),
            height: theme.spacing(6)
        }
    })
);

interface FileListPanelProps {
    rows: RowData[];
    selectedIndex: number;
    onSelectItemClick: (selectedIndex: number) => void;
}

const FileListPanel = (props: FileListPanelProps) => {
    const classes = useStyles();

    const handleListItemClick = (event: React.MouseEvent<HTMLDivElement, MouseEvent>, index: number) => {
        props.onSelectItemClick(index);
    };

    console.log(`Rendering FileListPanel ${JSON.stringify({ ...props, rows: props.rows.length })}`);
    return (
        <List className={classes.root}>
            {props.rows.map((row: RowData, index) => {
                return (
                    <React.Fragment key={row.title}>
                        <ListItem id={`listItem#${index}`} alignItems="flex-start" button selected={props.selectedIndex === index} onClick={(event) => handleListItemClick(event, index)}>
                            <ListItemAvatar>
                                <Avatar alt={row.title} src={row.thumbnail} variant="rounded" className={classes.large} />
                            </ListItemAvatar>
                            <ListItemText
                                primary={row.title}
                                secondary={
                                    <React.Fragment>
                                        <Typography component="span" variant="body2" className={classes.inline} color="textPrimary">
                                            {row.artist}
                                        </Typography>
                                        {row.album} [{row.year}]
                                    </React.Fragment>
                                }
                            />
                        </ListItem>
                        {index < props.rows.length && <Divider variant="inset" component="li" />}
                    </React.Fragment>
                );
            })}
        </List>
    );
};

export default FileListPanel;