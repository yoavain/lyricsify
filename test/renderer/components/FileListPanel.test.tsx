import * as React from "react";
import { create, act, ReactTestRenderer } from "react-test-renderer";
import FileListPanel from "../../../src/renderer/components/FileListPanel";
import { rows } from "../../resources/staticData";

describe("FileListPanel component", () => {
    it("renders correctly", () => {
        let root: ReactTestRenderer | undefined;
        act(() => {
            root = create(<FileListPanel rows={rows} onSelectItemClick={() => {}} selectedIndex={-1}/>);
        });

        expect(root?.toJSON()).toMatchSnapshot();

        act(() => {
            root?.update(<FileListPanel rows={rows} onSelectItemClick={() => {}} selectedIndex={0}/>);
        });
        expect(root?.toJSON()).toMatchSnapshot();
    });
});
