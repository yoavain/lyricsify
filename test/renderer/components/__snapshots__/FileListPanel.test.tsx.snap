// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileListPanel component renders correctly - none selected 1`] = `
<body>
  <div>
    <ul
      class="MuiList-root makeStyles-root-1 MuiList-padding"
    >
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#0"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-3"
          >
            <img
              alt="Happens To The Heart"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Happens To The Heart
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#1"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-3"
          >
            <img
              alt="Moving On"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Moving On
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#2"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-3"
          >
            <img
              alt="The Night of Santiago"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            The Night of Santiago
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#3"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-3"
          >
            <img
              alt="Thanks For The Dance"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Thanks For The Dance
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#4"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-3"
          >
            <img
              alt="It's Torn"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            It's Torn
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#5"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-3"
          >
            <img
              alt="The Goal"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            The Goal
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#6"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-3"
          >
            <img
              alt="Puppets"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Puppets
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#7"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-3"
          >
            <img
              alt="The Hills"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            The Hills
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#8"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-3"
          >
            <img
              alt="Listen To The Hummingbird"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Listen To The Hummingbird
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
    </ul>
  </div>
</body>
`;

exports[`FileListPanel component renders correctly - with selected 1`] = `
<body>
  <div>
    <ul
      class="MuiList-root makeStyles-root-4 MuiList-padding"
    >
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#0"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-6"
          >
            <img
              alt="Happens To The Heart"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Happens To The Heart
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-5 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#1"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-6"
          >
            <img
              alt="Moving On"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Moving On
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-5 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#2"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-6"
          >
            <img
              alt="The Night of Santiago"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            The Night of Santiago
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-5 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#3"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-6"
          >
            <img
              alt="Thanks For The Dance"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Thanks For The Dance
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-5 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#4"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-6"
          >
            <img
              alt="It's Torn"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            It's Torn
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-5 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#5"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-6"
          >
            <img
              alt="The Goal"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            The Goal
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-5 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#6"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-6"
          >
            <img
              alt="Puppets"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Puppets
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-5 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#7"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-6"
          >
            <img
              alt="The Hills"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            The Hills
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-5 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart"
        id="listItem#8"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-large-6"
          >
            <img
              alt="Listen To The Hummingbird"
              class="MuiAvatar-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Listen To The Hummingbird
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-5 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Leonard Cohen
            </span>
            Thanks For The Dance
             [
            2019
            ]
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <li
        class="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
    </ul>
  </div>
</body>
`;
